backend/controllers/board.js

```diff
@@ -66,6 +67,25 @@ boardRouter.get('/getBoard', async (req,res) => {
 });
 
 
+boardRouter.get('/getBoardsbyBook', async (req, res) => {
+    try{
+        const {userId, bookTitle, bookAuthor} = req.query;
+        const boards = await Board.find({
+            userId: { $ne: userId }, // Exclude boards that belong to the user
+            bookTitle: bookTitle,
+            bookAuthor: bookAuthor,
+            visibility: true
```

Comment:
could you update the "visibility" field to be "publicVisibility"? I think it's a little more clear that way